modules/core/src/lib/attribute.js

```diff
@@ -295,6 +301,12 @@ export default class Attribute extends BaseAttribute {
   }
 
   // PRIVATE HELPER METHODS
+  _getVertexOffset(index, bufferLayout = this.bufferLayout) {
+    if (bufferLayout) {
+      return bufferLayout.reduce((sum, cur, idx) => (idx < index ? sum + cur : sum), 0) * this.size;
```

Comment:
Oh no, a `reduce` statement, and with abbreviated parameter names... ;) 

I just find it much harder to verify at a glance that a `reduce` calculation is correct compared to a simple `for` loop.

Plus it looks like it always iterates to the end even though it doesn't have to?